### ISSUE TITLE
Sparkpool - Ethash - Region keeps changing + f2pool eth wallet problem.

### ISSUE THREAD

**Issue by khayto** [RainbowMiner/RainbowMiner#830]
on sparkpool every couple minutes miner restarts on a new region, JP than KR than CN. it was doing it on all miners, with or without dual mining ckb on nicehash, and i have no clue where its coming from, it doesnt behave like this for nicehash ethash or f2pool ethash.

speaking of f2pool, it seems theres 2 way to mine to it with some coins, the ones in the account wallet. eth for instance, if the user field in the bat file is a wallet address, its like an external wallet on nicehash, and if the field is an account name, you have to enter a wallet address in the payout address section on f2pool website. even if they are the same address, it will still count as 2 separate wallets. also i was wondering if using an account shouldnt also require a view only key generated from the account section of f2pool site in order for rbm to track balance. (unless you can get all wallets addresses associated to an account only using account name?)

long story short, the pool config for f2pool could use an additional field, possibly per coin listed on the website (imnt sure if 1 key gives access to all workers or just 1 coin) for view only key, those using the account name field should keep the wallet field empty and use only the view only key, while those using an external wallet should only fill the wallet field. 

example:
 "F2pool": {
    "UserName": "JOHN",
    "XZC": "a6QCxxxxxxxxxxxxxxxxxx",
    "XZC-Params": "",
    "ETH": "",
    "ETH-Params": "",
    "ETH-VOK": "xxxxx-xxxxxxxxxxx-xxxxxx",
    "ETC": "$ETC",
    "ETC-Params": "",
    "Worker": "$WorkerName",
    "Penalty": 6,
    "Algorithm": "MTP,ethash",
    "ExcludeAlgorithm": "",

**Comment by RainbowMiner** [comment]
Which Region have you setup in the config.txt?

**Comment by khayto** [comment]
US

**Comment by RainbowMiner** [comment]
Ok, found it. I'll push a fix.

**Comment by khayto** [comment]
thanks a lot!
